maps/time_key_map.go

```diff
@@ -0,0 +1,163 @@
+package maps
+
+import (
+	"time"
+
+	"github.com/akitasoftware/go-utils/sets"
+)
+
+func getInternalMapKey(t time.Time) int64 {
+	return t.Unix()
+}
+
+func getReverseKey(key int64) time.Time {
+	return time.Unix(key, 0)
+}
+
+type TimeMap[V any] struct {
+	internalMap map[int64]V
```

Comment:
**Blocking.** Let's avoid re-implementing `Map`. Each method in `TimeMap` should just be translation layer around a call to the corresponding method in `Map`.
```suggestion
	internalMap Map[int64, V]
```